engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -348,16 +349,28 @@ private void notifyJobAvailable(final DirectBuffer jobType) {
     }
   }
 
+  private void createJobRecord(final long key, final JobRecord record) {
+    jobKey.wrapLong(key);
+    // do not persist variables in job state
+    jobRecordToWrite.setRecordWithoutVariables(record);
+    jobsColumnFamily.insert(jobKey, jobRecordToWrite);
+  }
+
   private void resetVariablesAndUpdateJobRecord(final long key, final JobRecord updatedValue) {
     jobKey.wrapLong(key);
     // do not persist variables in job state
     jobRecordToWrite.setRecordWithoutVariables(updatedValue);
-    jobsColumnFamily.put(jobKey, jobRecordToWrite);
+    jobsColumnFamily.update(jobKey, jobRecordToWrite);
+  }
+
+  private void initializeJobState() {
```

Comment:
:question: Please also explain this change

Reply:
I've tried to explain this here: https://github.com/camunda-cloud/zeebe/pull/8802#discussion_r821897741